### PR TITLE
Instrument view handle invalid values better

### LINKED ISSUE #28787: InstrumentView not open for single bin
**Original reporter:** James Lord

### Expected behavior
Workspaces with Nans and Infs should open, even if they only have one bin
### Actual behavior
It does not open in this case
### Steps to reproduce the behavior
Run script:
```
LoadMuonNexus(Filename=r'\\chronus\data\CHRONUS00009705.nxs', OutputWorkspace='CHRONUS00009705.nxs', DeadTimeTable='CHRONUS00009705.nxs_deadtime_table', DetectorGroupingTable='__notUsed')
RenameWorkspace(InputWorkspace='CHRONUS00009705.nxs', OutputWorkspace='CHRONUS9705_raw_data MA')
 
LoadMuonNexus(Filename=r'\\chronus\data\CHRONUS00009706.nxs', OutputWorkspace='CHRONUS00009706.nxs', DeadTimeTable='CHRONUS00009706.nxs_deadtime_table', DetectorGroupingTable='__notUsed')
RenameWorkspace(InputWorkspace='CHRONUS00009706.nxs', OutputWorkspace='CHRONUS9706_raw_data MA')
 
Rebunch(InputWorkspace='CHRONUS9705_raw_data MA', OutputWorkspace='Integ9705', NBunch=5000)
Rebunch(InputWorkspace='CHRONUS9706_raw_data MA', OutputWorkspace='Integ9706', NBunch=5000)
Divide(LHSWorkspace='Integ9706', RHSWorkspace='Integ9705', OutputWorkspace='IntegRatio')
```
Then Right-click > show instrument on the `IntegRatio` Workspace
### Platforms affected
Found on Workbench 5.0.202020522.1617

### LINKED ISSUE #28788: Instrument view: Negative values not displayed
**Original reporter:** James Lord

### Expected behavior
The data (even negative) is represented by the colorbar
### Actual behavior
Any detectors with negative data values are greyed out, despite the colorbar trying to fit the correct range.
### Steps to reproduce the behavior
```
LoadMuonNexus(Filename=r'\\chronus\data\CHRONUS00009705.nxs', OutputWorkspace='CHRONUS00009705.nxs', DeadTimeTable='CHRONUS00009705.nxs_deadtime_table', DetectorGroupingTable='__notUsed')
RenameWorkspace(InputWorkspace='CHRONUS00009705.nxs', OutputWorkspace='Raw1')
 
LoadMuonNexus(Filename=r'\\chronus\data\CHRONUS00009706.nxs', OutputWorkspace='CHRONUS00009706.nxs', DeadTimeTable='CHRONUS00009706.nxs_deadtime_table', DetectorGroupingTable='__notUsed')
RenameWorkspace(InputWorkspace='CHRONUS00009706.nxs', OutputWorkspace='Raw2')
 
Minus(LHSWorkspace='Raw1', RHSWorkspace='Raw2', OutputWorkspace='Subtracted')
```
Right-click and Show Instrument on the `Subtracted` workspace

### Platforms affected
Found on Workbench 5.0.20200522.1617

### PR DESCRIPTION
**Pull request by NickDraper**
**Description of work.**
Read the commit messages.

**Report to:** James Lord

**To test:**
``` python
from mantid.simpleapi import *
ws=LoadMuonNexus(Filename=r'CHRONUS00009705.nxs', OutputWorkspace='CHRONUS00009705.nxs', DeadTimeTable='CHRONUS00009705.nxs_deadtime_table', DetectorGroupingTable='__notUsed')
RenameWorkspace(InputWorkspace='CHRONUS00009705.nxs', OutputWorkspace='CHRONUS9705_raw_data MA')
 
LoadMuonNexus(Filename=r'CHRONUS00009706.nxs', OutputWorkspace='CHRONUS00009706.nxs', DeadTimeTable='CHRONUS00009706.nxs_deadtime_table', DetectorGroupingTable='__notUsed')
RenameWorkspace(InputWorkspace='CHRONUS00009706.nxs', OutputWorkspace='CHRONUS9706_raw_data MA')
 
Rebunch(InputWorkspace='CHRONUS9705_raw_data MA', OutputWorkspace='Integ9705', NBunch=5000)
Rebunch(InputWorkspace='CHRONUS9706_raw_data MA', OutputWorkspace='Integ9706', NBunch=5000)
Divide(LHSWorkspace='Integ9706', RHSWorkspace='Integ9705', OutputWorkspace='IntegRatio')

neg = Load("MAR11060")
neg = neg * -1
```
Let me know if you need the files

1. Start workbench, run the script
1. Open Instrument view on neg
1. Check it has plotted the negative values correctly and pick mode works
1. Open IntegRatio
1. See grey detectors where there are nan values.
1. In pick mode mouse over a few of them see N/A in the counts text, the miniplot will have nothing because this is a single bin workspace.

Fixes #28787 and Fixes #28788.

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: qt/widgets/instrumentview/src/InstrumentActor.cpp [hunk 1]
```diff
@@ -60,6 +60,8 @@ bool isPhysicalView() {
 const size_t InstrumentActor::INVALID_INDEX =
     std::numeric_limits<size_t>::max();
 double InstrumentActor::m_tolerance = 0.00001;
+const double InstrumentActor::INVALID_VALUE =
+    -std::numeric_limits<double>::max();
```
Comment (ConorMFinn) [inline qt/widgets/instrumentview/src/InstrumentActor.cpp:64]: Could this be defined as a `constexpr` in the header file? 

`static constexpr double INVALID_VALUE = -std::numeric_limits<double>::max();`

**Review by cailafinn — CHANGES_REQUESTED** [review]
Functionally works well, negative numbers and NaNs now appearing on the Intrument View. Just the one comment on the code.

File: qt/widgets/instrumentview/src/InstrumentActor.cpp (hunk 1 above)
Comment (NickDraper) [reply qt/widgets/instrumentview/src/InstrumentActor.cpp]: Yes, I've also moved a couple of other consts in that file as well.

**Review by cailafinn — APPROVED** [review]
Looks good to me now. NaNs and Infinite values appearing as grey points, negative values correctly represented. Nice work 👍

**Pull request merged**